### PR TITLE
Add basic page infrastructure

### PR DESCRIPTION
**Pull request by ohltyler**
### Description

This PR adds some baseline infrastructure for the core pages the application will have:
- use cases page for selecting pre-existing or custom use cases which will redirect to the workflow builder page
- workflows page for managing created workflows
- workflow builder page for creating the workflows

Demo:
[screen-capture (13).webm](https://github.com/opensearch-project/opensearch-ai-flow-dashboards/assets/62119629/78abedc2-d785-412f-9811-f6acd587a20d)

### Check List

- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
- [x] Commits are signed per the DCO using `--signoff`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by ohltyler** [comment]
Marking draft until rebasing after #27 is merged

**Review by owaiskazi19 — APPROVED** [review]
LGTM!

File: public/pages/workflows/components/workflow_list.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      direction: sortDirection,
+    },
+    enableAllColumns: false,
+    readOnly: false,
+  };
+
+  const pagination = {
+    pageIndex,
+    pageSize,
+    totalItemCount: items.length,
```
Comment (jackiehanyang) [inline public/pages/workflows/components/workflow_list.tsx:54]: in what scenario this items length could change? it looks like it's a fixed 1 length now as line 36

**Comment by jackiehanyang** [comment]
overall LGTM, just a curiosity question

**Review by jackiehanyang — APPROVED** [review]

File: public/pages/workflows/components/workflow_list.tsx (hunk 1 above)
Comment (ohltyler) [reply public/pages/workflows/components/workflow_list.tsx:54]: Yeah just a placeholder item for now. We'll fetch the created workflows dynamically just like the AD Detector list page.

**Pull request merged**